mrmustard/lab_dev/circuit_components.py

```diff
@@ -209,6 +217,39 @@ def to_fock_component(
             self.wires,
         )
 
+    def __add__(self, other: CircuitComponent):
+        r"""
+        Implements the addition between circuit components.
+        """
+        if self.wires != other.wires:
+            msg = "Cannot add components with different wires."
+            raise ValueError(msg)
+        rep = self.representation + other.representation
+        name = self.name if self.name == other.name else ""
+        return self._from_attributes(name, rep, self.wires)
+
+    def __mul__(self, other: Union[CircuitComponent, Scalar]):
+        r"""
+        Implements the multiplication with a scalar or another component for circuit components.
```

Comment:
I'm not familiar with the interpretation of a "CircuitComponent" times a "CircuitComponent".

Reply:
that's a good point. we need it for scalars for sure though.

Reply:
:+1: 
I'll remove it